### PR TITLE
fix connector timeout

### PR DESCRIPTION
**Pull request by thehesiod**
this ensures the connector timeout only accounts for the time trying to connect to the endpoint  

underlying aiohttp issue: https://github.com/aio-libs/aiohttp/issues/2648
fixes aiobotocore issue: https://github.com/aio-libs/aiobotocore/issues/451

### CODE REVIEW

**Comment by jettify** [comment]
Lets not rush patching aiohttp and give @asvetlov time to take a look on aiohttp issue.

**Comment by jettify** [comment]
Depends on timing we can decide move forward with this patch or wait a bit.

**Comment by jettify** [comment]
As for patch, looks good, not sure how to test it.

**Comment by thehesiod** [comment]
based on previous conversations regarding timeouts I doubt they'll fix it, but ya we can wait and see.   ya let me think about a test.

**Comment by thehesiod** [comment]
@jettify ok added test, took awhile to get right as we needed a custom endpoint

File: tests/mock_server.py [hunk 1]
```diff
@@ -1,27 +1,110 @@
...
+        self._shutdown_evt = asyncio.Event()
+
+    def _run(self):
+        self._loop = asyncio.new_event_loop()
+        app = aiohttp.web.Application(loop=self._loop)
+        app.router.add_route('*', '/ok', self.ok)
+        app.router.add_route('*', '/{anything:.*}', self.stream_handler)
+
+        try:
+            aiohttp.web.run_app(app, host='localhost', port=self._port,
```
Comment (jettify) [inline tests/mock_server.py]: you can run aiohttp server in same thread without using threading. See example here https://github.com/aio-libs/aiozipkin/blob/5f914c6eb811db046a58d5d816cd19223ae2435c/demos/queue/runner.py#L7-L25 it is a bit broken I wrote those late night, but you get the idea

File: tests/mock_server.py [hunk 2]
```diff
@@ -56,8 +139,11 @@ def s3_server():
     port = 5000
     url = "http://{host}:{port}".format(host=host, port=port)
     process = start_service('s3', host, port)
-    yield url
-    stop_process(process)
+
+    try:
+        yield url
+    finally:
```
Comment (jettify) [inline tests/mock_server.py]: 👍

File: tests/mock_server.py (hunk 1 above)
Comment (thehesiod) [reply tests/mock_server.py]: cool, want me migrate?

Comment (jettify) [reply tests/mock_server.py]: yes please.

**Comment by jettify** [comment]
I think timeouts will not be fixed in next major release, so we can move forward with your patch.

File: tests/mock_server.py (hunk 1 above)
Comment (thehesiod) [reply tests/mock_server.py]: i just realized I shutdown the server via `loop.stop` however we cannot do this in a unittest so I think we need to keep it in a thread.

**Comment by thehesiod** [comment]
getting this run on travis on 3.4 is not fun

**Comment by thehesiod** [comment]
coverage looks good, for some reason didn't update in this PR

**Pull request merged**